src/common/git/husky/with-git-hooks.ts

```diff
@@ -0,0 +1,47 @@
+export interface WithGitHooks {
+  /**
+   * Include husky for git hook management.
+   *
+   * @default false
+   */
+  readonly hasGitHooks?: boolean
+
+  /**
+   * Include a default git hook that checks commit message.
+   *
+   * @default { commitMsg: true }
+   */
+  readonly huskyRules?: HuskyRule
+}
+
+export interface HuskyRule {
+  /**
+   * Include pre-commit hook with `cargo check` command.
+   *
+   * @default undefined
+   */
+  readonly checkCargo?: CheckCargoOptions
```

Comment:
@quesabe if these types are only used in this file let's not export them. If not, let's put one type per file, as we usually do.